From 392b3a211f37a9fd80e60ec44cd6462e21cc2739 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Mon, 13 Jan 2014 10:26:41 +0100
Subject: [GNOME-ABRT PATCH 1/3] Do not crash when a FileIcon cant be loaded

Resolves rhbz#1052006
Resolves rhbz#1054089

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/gnome_abrt/application.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/gnome_abrt/application.py b/src/gnome_abrt/application.py
index 140042c..ca81f6c 100644
--- a/src/gnome_abrt/application.py
+++ b/src/gnome_abrt/application.py
@@ -149,8 +149,12 @@ def find_application(component, executable, cmdline):
                             except GLib.GError as ex:
                                 logging.debug(ex)
                     elif isinstance(dai_icon, Gio.FileIcon):
-                        stream = dai_icon.load(128, None)
-                        icon = GdkPixbuf.Pixbuf.new_from_stream(stream[0], None)
+                        try:
+                            stream = dai_icon.load(128, None)
+                            icon = GdkPixbuf.Pixbuf.new_from_stream(stream[0],
+                                                                    None)
+                        except GLib.GError as ex:
+                            logging.debug(ex)
                     else:
                         logging.debug("Unsupported type of icon class: {0}"
                                 .format(dai_icon))
-- 
1.8.3.1

